Issue backend CUDA with boost 1.69.0 and cuda 10

Hello, 
I want to use the cuda backend of vexcl but it doesn't work I have the following error :
 
`Error		explicit specialization of class "vex::traits::is_vector_expr_terminal<size_t, void>" must precede its first use  C:\Utils\vexcl-master\vexcl\backend\cuda\texture_object.hpp	41`

I have boost 1.69, visual studio 2017 15.9.7 and cuda 10. Maybe my boost version is too recent ?

I tried to compile the examples of vexcl as you said before but I have many errors like this one
`CMake Error at CMakeLists.txt:10 (add_executable):
  Target "complex_spmv" links to target "VexCL::Backend" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  CMakeLists.txt:27 (add_vexcl_examples)`

--- a/tests/image.cpp
+++ b/tests/image.cpp
@@ -11,7 +11,7 @@ BOOST_AUTO_TEST_CASE(image1d)
 {
     if (!vex::Filter::CC(3, 0)(ctx.device(0))) return;
 
-    VEX_FUNCTION(float, imread, (CUtexObject, tex)(int, i),
+    VEX_FUNCTION(float, imread, (vex::cuda_tex_object, tex)(int, i),
             return tex1Dfetch<float>(tex, i);
             );
 
@@ -41,7 +41,7 @@ BOOST_AUTO_TEST_CASE(image1d)
 
     vex::vector<float> x(q1, n);
 
-    x = imread(tex, p);
+    x = imread(vex::wrap(tex), p);
 
     check_sample(x, [](size_t, float a) {
             BOOST_CHECK_EQUAL(a, 42);

--- a/vexcl/backend/cuda/texture_object.hpp
+++ b/vexcl/backend/cuda/texture_object.hpp
@@ -37,11 +37,19 @@ THE SOFTWARE.
 
 namespace vex {
 
-namespace traits {                                                             \
-template <> struct is_vector_expr_terminal<CUtexObject> : std::true_type {};
+struct cuda_tex_object {
+    CUtexObject h;
+};
+
+inline cuda_tex_object wrap(CUtexObject h) {
+    return cuda_tex_object{h};
 }
 
-template <> struct type_name_impl<CUtexObject> {
+namespace traits {                                                             \
+template <> struct is_vector_expr_terminal<cuda_tex_object> : std::true_type {};
+}
+
+template <> struct type_name_impl<cuda_tex_object> {
     static std::string get() { return "cudaTextureObject_t"; }
 };
 

